### ISSUE TITLE
Static-only objects are included in wildcards expansions

### ISSUE THREAD

**Issue by pawamoy** [mkdocstrings/griffe#42]
**Describe the bug**
Currently, expanding a wildcard will include members that are only "statically" present.
More specifically: members that are defined or imported under an `if TYPE_CHECKING` or `if False` condition.
This is problematic because this can lead to cyclic aliases.

**To Reproduce**

```python
# package/module_a.py
from typing import TYPE_CHECKING

if TYPE_CHECKING:  # always false
    from package.module_b import CONST
```

```python
# package/__init__.py
from .module_a import *
```

At runtime, `package` would definitely *not* have `CONST` in its members.

**Expected behavior**
These "static-only" objects should not be included in wildcards expansions.

**Additional context**
https://github.com/mkdocstrings/mkdocstrings/issues/382
Maybe an other way to fix that would be to better handle duplicate members.
But that depends on the order we add expanded wildcard's members...

**Comment by tristanlatr** [comment]
This is how I check for the TYPE checking blocks: https://github.com/tristanlatr/pydocspec/blob/main/pydocspec/astroidutils.py#L359-L374

Objects have a field `is_type_guarded` wich is set to True is the context is under a type guard if condition. 

Then the wildcard import rely on `__all__` or the public names returnde with this function: https://github.com/tristanlatr/pydocspec/blob/main/pydocspec/processor/mod_attr.py#L75-L87

**Comment by pawamoy** [comment]
Thanks @tristanlatr! I was indeed thinking about adding a field on objects so that we know they are type-guarded or not.

**Comment by pawamoy** [comment]
Released in 0.17.0!
